### PR TITLE
SimCLR: A Simple Framework for Contrastive Learning of Visual Representations

### LINKED ISSUE #1224: SimCLR: A Simple Framework for Contrastive Learning of Visual Representations
### Brief Summary: 
Describe the what, why, and how of your content idea in 2-5 sentences.

Constrastive Learning date as far back as the 1990s. It is a learning paradigm where a representation is learned by comparing among the input samples. Instead of learning a signal from individual data samples one at a time, contrastive learning learns by comparing among different samples. 
In this article, we will learn about the SimCLR framework which is an improvement of previous contrastive learning approaches to self-supervised representation learning on images.

### Key Takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?

By the end of this article, the reader will be able to understand:

1. What is Contrastive Learning?
2. The SimCLR Framework
3. Applications of Contrastive Learning in Computer Vision, Natural Language Processing and Reinforcement Learning.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)

1. [Advancing Self-Supervised and Semi-Supervised Learning with SimCLR] (https://ai.googleblog.com/2020/04/advancing-self-supervised-and-semi.html) 
2. [A Simple Framework for Contrastive Learning of Visual Representations ] (https://arxiv.org/pdf/2002.05709.pdf)
3. [Contrastive Representation Learning: A Framework and Review] (https://arxiv.org/ftp/arxiv/papers/2010/2010.05113.pdf)

### PR DESCRIPTION
**Pull request by Bayler**
…ntations

This fixes #1224

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

File: articles/simclr-a-simple-framework-for-contrastive-learning-of-visual-representations/index.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
+### Prerequisites
+
+Before reading this article, the reader must understand Machine Learning (ML) and Deep Learning (DL). If you are still new to ML and DL, please read my previously published article on the [differences between Artificial Intelligence, Machine Learning, and Deep Learning](/engineering-education/differences-between-artificial-intelligence-machine-learning-and-deep-learning/) to learn more.
+
+### What is Contrastive Learning?
+
+Contrastive learning is a very active area in machine learning research. It is a self-supervised method used in machine learning to put together the task of finding similar and dissimilar things. By applying this method, one can train a machine learning model to contrast similar and dissimilar images. For example, given an image of a horse, one can find the matching animal in a gallery of other photos.
```
Comment (lalith1403) [inline articles/simclr-a-simple-framework-for-contrastive-learning-of-visual-representations/index.md:7]: An introduction to self-supervised learning is required.

File: articles/simclr-a-simple-framework-for-contrastive-learning-of-visual-representations/index.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+SimCLR is a framework developed by Google which demonstrated the implications of contrastive learning. It is a high-impact work that eliminates specialized architectures and memory banks, usually used in contrastive learning. It shows that strong augmentations of unlabeled training data, a standard ResNet-50 architecture, and a small neural network is all you need to achieve state-of-the-art results. For such a simple approach, the results are truly mindblowing.
```
Comment (lalith1403) [inline articles/simclr-a-simple-framework-for-contrastive-learning-of-visual-representations/index.md:9]: Insert links to data augmentation, Resnet-50.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Very well explained. Can we come up with code to implement SimCLR framework? That would be amazing (in another article). In this article, there are a few places where more information and links are required. Add them wherever you feel more information is necessary. I have highlighted a few such places.

**Comment by Bayler** [comment]
> 
> 
> Very well explained. Can we come up with code to implement SimCLR framework? That would be amazing (in another article). In this article, there are a few places where more information and links are required. Add them wherever you feel more information is necessary. I have highlighted a few such places.

@lalith1403  Sure. That would be amazing. I'll do that in my next article as a follow-up. Let me work on the few revisions

**Comment by hectorkambow** [comment]
Plagiarism check done

**Comment by Bayler** [comment]
@lalith1403 finished with the revisions

**Pull request merged**